File: lc/math/reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        if (x == INT_MAX || x == INT_MIN) {
            return 0;
        }
        int negative = 1;
        if (x < 0) {
            x = -x;
            negative = -1;
        }
        int reversed_x = 0;
        while (x > 0) {
            int d = x % 10;
            if(reversed_x > (INT_MAX-d) / 10) {
                return 0;
            }
            reversed_x = reversed_x * 10 + d;
            x /= 10;
        }
        std::cout << negative << std::endl;
        return negative * reversed_x;
    }
};

File: lc/math/reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        pre = 1
        if x < 0:
            x = -x
            pre = -1

        x = -x if x < 0 else x
        reversed_x = 0
        bound = (1 << 31) - 1

        while x > 0:
            d = x % 10
            reversed_x = reversed_x * 10 + d
            if reversed_x >= bound:
                return 0
            x //=10

        return pre * reversed_x
